homeassistant/components/workday/config_flow.py

```diff
@@ -0,0 +1,302 @@
+"""Adds config flow for Workday integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import holidays
+from holidays import HolidayBase
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    OptionsFlowWithConfigEntry,
+)
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import AbortFlow, FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.selector import (
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+    TextSelector,
+)
+from homeassistant.util import dt
+
+from .const import (
+    ALLOWED_DAYS,
+    CONF_ADD_HOLIDAYS,
+    CONF_COUNTRY,
+    CONF_EXCLUDES,
+    CONF_OFFSET,
+    CONF_PROVINCE,
+    CONF_REMOVE_HOLIDAYS,
+    CONF_WORKDAYS,
+    DEFAULT_EXCLUDES,
+    DEFAULT_NAME,
+    DEFAULT_OFFSET,
+    DEFAULT_WORKDAYS,
+    DOMAIN,
+)
+
+NONE_SENTINEL = "none"
+
+
+def add_province_to_schema(
+    schema: vol.Schema,
+    options: dict[str, Any],
+) -> vol.Schema:
+    """Update schema with province from country."""
+    year: int = dt.now().year
+    obj_holidays: HolidayBase = getattr(holidays, options[CONF_COUNTRY])(years=year)
+    if not obj_holidays.subdivisions:
+        return schema
```

Comment:
Looks good, but this needs coverage now.

Reply:
```
---------- coverage: platform linux, python 3.10.10-final-0 ----------
Name                                                Stmts   Miss  Cover   Missing
---------------------------------------------------------------------------------
homeassistant/components/workday/__init__.py           12      0   100%
homeassistant/components/workday/binary_sensor.py     102      0   100%
homeassistant/components/workday/config_flow.py       103      0   100%
homeassistant/components/workday/const.py              18      0   100%
---------------------------------------------------------------------------------
TOTAL                                                 235      0   100%
```